commands/src/main/java/org/wildfly/extras/creaper/commands/deployments/Deploy.java

```diff
@@ -0,0 +1,211 @@
+    private static final class Data {
+
+        private File deploymentFile;
+        private InputStream deploymentInputStream;
+        private final boolean autoCloseInputStream;
+        private final String deploymentName;
+
+        private List<String> serverGroups = null;
+
+        public Data(InputStream deploymentInputStream, String deploymentName, boolean autoCloseInputStream) {
+            this.deploymentInputStream = deploymentInputStream;
+            this.deploymentName = deploymentName;
+            this.autoCloseInputStream = autoCloseInputStream;
+        }
+
+        public Data(File deploymentFile) {
+            if (!deploymentFile.exists()) {
+                throw new IllegalArgumentException(deploymentFile + " doesn't exist!");
+            }
+            this.deploymentFile = deploymentFile;
+            this.deploymentName = deploymentFile.getName();
+            this.autoCloseInputStream = true;
+        }
+
+        public Deploy build() {
+            InputStream inputStream = deploymentInputStream;
+            if (deploymentInputStream == null && deploymentFile != null) {
+                try {
+                    inputStream = new FileInputStream(deploymentFile);
+                } catch (FileNotFoundException ex) {
+                    throw new IllegalArgumentException(ex);
+                }
+            }
+            return new Deploy(inputStream, this);
+        }
+    }
+
+    /**
+     * Builder for creating command for deploying deployment to the server.
+     */
+    public static final class DeployCmdBuilder {
+        private final Data deployCmdData;
+
+        public DeployCmdBuilder(final File deploymentFile) {
+            this.deployCmdData = new Data(deploymentFile);
+        }
+
+        public DeployCmdBuilder(final InputStream deploymentInputStream, final String deploymentName,
+                                boolean autoCloseInputStream) {
+            this.deployCmdData = new Data(deploymentInputStream, deploymentName, autoCloseInputStream);
+        }
+
+        public Deploy build() {
+            return deployCmdData.build();
+        }
+
+        /**
+         * Allows to specify additional options which are specific to server groups in domain.
+         *
+         * @return builder for specifying specific options related to server groups in domain.
+         */
+        public ServerGroupOptions serverGroupOptions() {
+            return new ServerGroupOptions(deployCmdData);
+        }
+    }
+
+    /**
+     * Builder adding domain specific options for the deploy command
+     */
+    public static final class ServerGroupOptions {
+
+        private final Data deployCmdData;
+
+        private ServerGroupOptions(Data deployCmdData) {
+            this.deployCmdData = deployCmdData;
+        }
+
+        /**
+         * Adds list of server groups for which the deployment should be applied.
+         */
+        public Deploy toParticularServerGroups(String... serverGroups) {
```

Comment:
I think `toServerGroups` would be shorter and better (more symmetric with `toAllServerGroups`).